src/main/java/edu/cornell/kfs/module/cam/document/service/impl/CuAssetGlobalServiceImpl.java

```diff
@@ -19,5 +25,42 @@ protected Asset setupAsset(final AssetGlobal assetGlobal, final AssetGlobalDetai
         return asset;
 
     }
+    
+    @Override
+    protected void setupAssetLocationOffCampus(final AssetGlobalDetail assetGlobalDetail, final Asset asset) {
+        // We are not checking if it already exists since on a new asset it can't
+        final AssetLocation offCampusAssetLocation = new AssetLocation();
+        offCampusAssetLocation.setCapitalAssetNumber(asset.getCapitalAssetNumber());
+        offCampusAssetLocation.setAssetLocationTypeCode(CamsConstants.AssetLocationTypeCode.OFF_CAMPUS);
+        asset.getAssetLocations().add(offCampusAssetLocation);
+
+        // Set the location fields either way
+        offCampusAssetLocation.setAssetLocationContactName(assetGlobalDetail.getOffCampusName());
+        offCampusAssetLocation.setAssetLocationContactIdentifier(assetGlobalDetail.getRepresentativeUniversalIdentifier());
+        processAssetLocationInstitutionName(assetGlobalDetail, offCampusAssetLocation);
+        offCampusAssetLocation.setAssetLocationStreetAddress(assetGlobalDetail.getOffCampusAddress());
+        offCampusAssetLocation.setAssetLocationCityName(assetGlobalDetail.getOffCampusCityName());
+        offCampusAssetLocation.setAssetLocationStateCode(assetGlobalDetail.getOffCampusStateCode());
+        offCampusAssetLocation.setAssetLocationCountryCode(assetGlobalDetail.getOffCampusCountryCode());
+        offCampusAssetLocation.setAssetLocationZipCode(assetGlobalDetail.getOffCampusZipCode());
+
+        // There is no phone number field on Asset Global... odd...
+        offCampusAssetLocation.setAssetLocationPhoneNumber(null);
+    }
+    
+    /*
+     * CU-Customization
+     * The asset representative might mot be a Cornell person, so we need to handle this situation
+     * KualiCo jira FINP-11058 should address this bug
+     */
+    private void processAssetLocationInstitutionName(final AssetGlobalDetail assetGlobalDetail,
+            final AssetLocation offCampusAssetLocation) {
+        if (ObjectUtils.isNotNull(assetGlobalDetail.getAssetRepresentative())) {
+            offCampusAssetLocation.setAssetLocationInstitutionName(
+                    assetGlobalDetail.getAssetRepresentative().getPrimaryDepartmentCode());
+        } else {
+            LOG.info("processAssetLocationInstitutionName, the asset representative is not sent, unable to set asset location institution name");
```

Comment:
Should this message be changed from `...not sent...` to `...not set...`?

Reply:
ok